src/main/java/io/cdap/plugin/hive/action/HiveImport.java

```diff
@@ -40,24 +41,21 @@ public HiveImport(HiveConfig config) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
-    validateImportConfig();
-  }
-
-  private void validateImportConfig() {
-    if (!config.containsMacro("statement")) {
-      // Load command should not allow local storage
-      if (config.statement.length() >= 15 && config.statement.substring(10, 15).equalsIgnoreCase("LOCAL")) {
-        throw new IllegalArgumentException("Hive Import does not allow local file storage." +
-                                             "Please import data to HDFS location.");
-      }
-    }
+      FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+      config.validate(failureCollector);
+      config.validateImportStatment(failureCollector);
   }
 
   @Override
   public void run(ActionContext context) throws Exception {
-    validateImportConfig();
-    HiveCommandExecutor executor = new HiveCommandExecutor(config.connectionString, config.user, config.password);
-    executor.execute(config.statement);
+    FailureCollector failureCollector = context.getFailureCollector();
```

Comment:
why are we doing this in run instead of prepareRun? Does Action not have prepareRun?

Reply:
Action doesn't have prepareRun method.